RadClass/RadClass.py

```diff
@@ -171,17 +195,17 @@ def iterate(self):
         Only runs for a set node (datapath) with data already queued.
         '''
         bar = progressbar.ProgressBar(max_value=100, redirect_stdout=True)
-        inverse_dt = 1.0 / (self.processor.timestamps[-1]
-                            - self.processor.timestamps[0])
+        inverse_dt = 1.0 / (self.stop_i - self.start_i)
+        offset = self.start_i
 
         log_interval = 10000  # number of samples analyzed between log updates
         running = True  # tracks whether to end analysis
         while running:
             # print status at set intervals
-            if np.where(self.processor.timestamps == self.working_time)[0][0] % log_interval == 0:
-                bar.update(round((self.working_time - self.processor.timestamps[0]) * inverse_dt, 4)*100)
+            if (self.start_i-offset) % log_interval == 0:
```

Comment:
concern for future you: I have mixed feelings about `start_i` now really meaning `current_i`.  I know you felt that `working_time` was superfluous, but semantically it is useful to distinguish between the start of your domain and the place where you are currently starting an integration. There may be a time in the future where you need to know these things separately, and it also makes the code more readable to know where the current index is.

Reply:
I agree. I have added a `current_i` and separated it from `start_i`. I noticed that this was needed when I was creating the progressbar `offset`, since the actual start was overwritten throughout the course of processing.